Sources/StreamChat/Repositories/MessageRepository.swift

```diff
@@ -165,15 +165,19 @@ class MessageRepository {
     /// - Parameters:
     ///   - cid: The channel identifier the message relates to.
     ///   - messageId: The message identifier.
+    ///   - store: A boolean indicating if the message should be stored to database or should only be retrieved
     ///   - completion: The completion. Will be called with an error if something goes wrong, otherwise - will be called with `nil`.
-    func getMessage(cid: ChannelId, messageId: MessageId, completion: ((Result<ChatMessage, Error>) -> Void)? = nil) {
+    func getMessage(cid: ChannelId, messageId: MessageId, store: Bool = true, completion: ((Result<ChatMessage, Error>) -> Void)? = nil) {
         let endpoint: Endpoint<MessagePayload.Boxed> = .getMessage(messageId: messageId)
         apiClient.request(endpoint: endpoint) {
             switch $0 {
             case let .success(boxed):
                 var message: ChatMessage?
                 self.database.write({ session in
                     message = try session.saveMessage(payload: boxed.message, for: cid, syncOwnReactions: true, cache: nil).asModel()
+                    if !store {
+                        self.database.writableContext.discardCurrentChanges()
```

Comment:
Completely aligned. But as of now, we don't have a way to convert a payload to a particular entity. We only have payload -> model -> entity. Our entities even require a NSManagedObjectContext because of the usage of CoreDataLazy. Therefore, our current infrastructure does not allow for the straight path that we want here.

Again, this is something that is not ideal.